pkg/controller/razeedeployment/razeedeployment_controller.go

```diff
@@ -208,259 +180,270 @@ func (r *ReconcileRazeeDeployment) Reconcile(request reconcile.Request) (reconci
 	}
 
 	/******************************************************************************
-	CHECK THE INSTANCE FOR VALUES PASSED DOWN FROM MARKETPLACE CONFIG
-	check the instance for rhmSecretNameNonNil
-	check the instance for *clusterUUID
+	PROCEED WITH CREATING RAZEE PREREQUISITES?
 	/******************************************************************************/
-	rhmSecretName := "rhm-operator-secret"
-	clusterUUID := &instance.Spec.ClusterUUID
-	targetNamespace := *instance.Spec.TargetNamespace
-
-	if instance.Spec.DeploySecretName != nil {
-		rhmSecretName = *instance.Spec.DeploySecretName
+	if instance.Spec.DeployConfig == nil {
+		reqLogger.Info("rhm-operator-secret has not been applied")
+		return r.reconcileRhmOperatorSecret(*instance, request)
 	}
 
-	/******************************************************************************
-	CHECK FOR COMBINED SECRET
-	check for the presence of the combined secret
-	/******************************************************************************/
-	combinedSecret := corev1.Secret{}
-	err = r.client.Get(context.TODO(), types.NamespacedName{
-		Name:      rhmSecretName,
-		Namespace: request.Namespace,
-	}, &combinedSecret)
-	if err != nil {
-		if errors.IsNotFound(err) {
-			// Request object not found, could have been deleted after reconcile request.
-			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
-			// Return and don't requeue
-			// report to status that we haven't found the secret
-			reqLogger.Info("Updating RedHatMarketplaceSecretFound")
-			instance.Status.RedHatMarketplaceSecretFound = &redHatMarketplaceSecretFound
-			*instance.Status.RedHatMarketplaceSecretFound = false
-			err = r.client.Status().Update(context.TODO(), instance)
+	// TODO: is there a way to refactor this so reconcileRhmOperatorSecret() doesn't have to be called twice.
+	if instance.Spec.DeployConfig != nil {
+		if len(instance.Status.MissingDeploySecretValues) > 0 {
+			reqLogger.Info("Missing required razee configuration values")
+			return r.reconcileRhmOperatorSecret(*instance, request)
+		} else {
+			reqLogger.Info("all secret values found")
+
+			//construct the childURL
+			url := fmt.Sprintf("%s/%s/%s/%s", instance.Spec.DeployConfig.IbmCosURL, instance.Spec.DeployConfig.BucketName, instance.Spec.ClusterUUID, instance.Spec.DeployConfig.ChildRSSFIleName)
+			instance.Spec.ChildUrl = &url
+			err = r.client.Update(context.TODO(), instance)
 			if err != nil {
-				reqLogger.Error(err, "Failed to update Status.RedHatMarketplaceSecretFound")
+				reqLogger.Error(err, "Failed to update ChildUrl")
+				return reconcile.Result{}, err
```

Comment:
We can move this logic into the end of the reconcileRhmOperatorSecret.

Reply:
Thought about doing that. I was worried about having a partially formed url. As is, I know all the values are there and can create a full url.

Reply:
If you don't have any missing fields and you recalculate it each time the secret is found then by the time you get here the child url will be complete.